app/Http/Controllers/PermissionController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Spatie\Permission\Models\Role;
+use Spatie\Permission\Models\Permission;
+
+class PermissionController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('role:admin');
+    }
+    
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $permissions = Permission::all();
+        return response()->json(['status' => 'success', 'permissions' => $permissions]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name'=>'required|max:40',
+        ]);
+
+        $name = $request->input('name');
+        $permission = new Permission();
+        $permission->name = $name;
+        $permission->save();
+
+        $roles = $request->input('roles');
+        if (!empty($roles)) {
+            foreach ($roles as $role) {
+                $dbRole = Role::where('id', '=', $role)->first();
+                if ($dbRole) {
+                    $dbPerm = Permission::find($permission->id);
+                    $dbRole->givePermissionTo($dbPerm);
+                } else {
+                    return response()->json(['status' => 'error', 'message' => "Role $role not found."], 422);
+                }
+            }
+        }
+        
+        $dbPermission = Permission::find($permission->id);
```

Comment:
permissions also has a `findByName()` method.

Reply:
This pulls from the permission that was just created, no need for a name there since we already have the `id`.